Providers/Scripts/2.6x-2.7x/Scripts/nxOMSAutomationWorker.py

```diff
@@ -232,37 +152,111 @@ def Get_Marshall(WorkspaceId, Enabled, AzureDnsAgentSvcZone):
+OPTION_WORKSPACE_ID = "workspace_id"
+
+
+SECTION_OMS_WORKER_CONF = "oms-worker-conf"
+OPTION_RESOURCE_VERSION = "resource_version"
+OPTION_HYBRID_WORKER_PATH = "hybrid_worker_path"
+OPTION_DISABLE_WORKER_CREATION = "disable_worker_creation"
+
 
 WORKER_STATE_DIR = "/var/opt/microsoft/omsagent/state/automationworker"
-WORKER_CONF_FILE_PATH = os.path.join(WORKER_STATE_DIR, "worker.conf")
-WORKER_STATE_FILE_PATH = os.path.join(WORKER_STATE_DIR, "state.conf")
+DIY_WORKER_STATE_DIR = os.path.join(WORKER_STATE_DIR, "diy")
+OMS_CONF_FILE_PATH = os.path.join(WORKER_STATE_DIR, "oms.conf")
+AUTO_REGISTERED_WORKER_CONF_PATH = os.path.join(WORKER_STATE_DIR, "worker.conf")
+MANUALLY_REGISTERED_WORKER_CONF_PATH = os.path.join(DIY_WORKER_STATE_DIR, "worker.conf")
+STATE_CONF_FILE_PATH = os.path.join(WORKER_STATE_DIR, "state.conf")
 
 DSC_RESOURCE_VERSION_FILE = "/opt/microsoft/omsconfig/modules/nxOMSAutomationWorker/VERSION"
 OMS_ADMIN_CONFIG_FILE = "/etc/opt/microsoft/omsagent/conf/omsadmin.conf"
-OMS_CERTIFICATE_PATH = "/etc/opt/microsoft/omsagent/certs/oms.crt"
-OMS_CERT_KEY_PATH = "/etc/opt/microsoft/omsagent/certs/oms.key"
 WORKING_DIRECTORY_PATH = "/var/opt/microsoft/omsagent/run/automationworker"
-REGISTRATION_FILE_PATH = "/opt/microsoft/omsconfig/modules/nxOMSAutomationWorker/DSCResources/MSFT_nxOMSAutomationWorkerResource/automationworker/scripts/register_oms.py"
-HYBRID_WORKER_START_PATH = "/opt/microsoft/omsconfig/modules/nxOMSAutomationWorker/DSCResources/MSFT_nxOMSAutomationWorkerResource/automationworker/worker/main.py"
+WORKER_MANAGER_START_PATH = "/opt/microsoft/omsconfig/modules/nxOMSAutomationWorker/DSCResources/MSFT_nxOMSAutomationWorkerResource/automationworker/worker/main.py"
+HYBRID_WORKER_START_PATH = "/opt/microsoft/omsconfig/modules/nxOMSAutomationWorker/DSCResources/MSFT_nxOMSAutomationWorkerResource/automationworker/worker/hybridworker.py"
 PROXY_CONF_PATH_LEGACY="/etc/opt/microsoft/omsagent/conf/proxy.conf"
 PROXY_CONF_PATH_NEW="/etc/opt/microsoft/omsagent/proxy.conf"
+REGISTRATION_FILE_PATH = "/opt/microsoft/omsconfig/modules/nxOMSAutomationWorker/DSCResources/MSFT_nxOMSAutomationWorkerResource/automationworker/scripts/register_oms.py"
+OMS_CERTIFICATE_PATH = "/etc/opt/microsoft/omsagent/certs/oms.crt"
+OMS_CERT_KEY_PATH = "/etc/opt/microsoft/omsagent/certs/oms.key"
 KEYRING_PATH="/etc/opt/omi/conf/omsconfig/keyring.gpg"
-LOCAL_LOG_LOCATION = "/var/opt/microsoft/omsagent/log/nxOMSAutomationWorker.log"
 
 # permission level rwx rwx ---
 # leading zero is necessary because this is an octal number
 # Note: for python 3.x use 0o770 instead of 0770
 PERMISSION_LEVEL_0770 = 0770
+PERMISSION_LEVEL_0777 = 0777
 
 AUTOMATION_USER = "nxautomation"
 
+LOCAL_LOG_LOCATION = "/var/opt/microsoft/omsagent/log/nxOMSAutomationWorker.log"
+LOG_LOCALLY = False
+
+# User defined functions
+
+def read_settings_from_mof_json(json_serialized_string):
+    try:
+        json_serializer = serializerfactory.get_serializer(sys.version_info)
```

Comment:
This is encapsulated in a single method but the serialize could be set at the global scope (if other methods have to use it).